_posts/2022-11-20-point-in-time.md

```diff
@@ -0,0 +1,232 @@
+
+## OpenSearch pagination methods 
+
+There are four [ways to paginate results](https://opensearch.org/docs/latest/opensearch/search/paginate/) in OpenSearch:
+
+- The `from` and `size` parameters
+- The scroll search operation
+- The `search_after` parameter
+- PIT with `search_after`
+
+### What makes a good pagination method?
+
+So, what are the desired characteristics of a good pagination method? Of course, it depends on your application. If you don't need the ability to skip pages, you may be perfectly fine with the most basic scroll search. However, in general here are the qualities of a good pagination method:
+
+- **Moving forward and backward**: In addition to moving forward in search results, the user may want to go back to the page before the current page.
+- **Skipping pages**: The user may want to skip to a page out of order.
+- **Consistency**: The search results must stay consistent even with live index updates. If the user is on Page 1 of the results and selects Page 2 and then goes back to Page 1, Page 1 stays the same despite documents being indexed or deleted in the meantime.
+- **Deep pagination**: The search must stay efficient even when the user wants to view results starting with Page 1000.
+
+### Pagination methods, compared
+
+Here is how OpenSearch pagination methods compare to each other.
+
+<style>
+table{
+    border:2px solid #e6e6e6;
+}
+
+th{
+    border:2px solid #e6e6e6;
+    padding: 5px;
+    text-align: center;
+}
+
+td{
+    border:1px solid #e6e6e6;
+    padding: 10px;
+    text-align: center;
+}
+</style>
+
+Pagination method | Can move forward and backward? | Consistent pagination? | Efficient for deep pagination? | Results not bound to a particular query? | 
+:--- | :--- | :--- | :--- | :--- |
+**`from` and `size` parameters** | &#10004; | - | - | - |
+**Scroll search** | - | &#10004; | - | - |
+**`search_after` parameter** | &#10004; | - | - | - |
+**PIT with `search_after`** | &#10004; | &#10004; | &#10004; | &#10004; |
+
+### And the winner is...
+
+As you can see, PIT with `search_after` is a clear winner because it checks all the boxes. Not only can it move forward and backward in search results and provide consistent pagination while documents are being indexed and deleted, but it is also efficient for deep pagination. Plus, the results are frozen in time and not bound to a particular query, so you can run different queries against the same result dataset.
+
+## How PIT works
+
+When you create a PIT for a set of indexes, OpenSearch takes the corresponding segments of the indexes' shards and freezes them in time, creating contexts (pointers to the data) that you can use to access and query those shards. 
```

Comment:
See global comment. Should this read "When you create PIT for.." or "When you create a PIT search...?"

Reply:
create a PIT as a physical thing... so "a" is necessary. Like "create a program". And you don't create a search.